commerce-inventory-test/src/testIntegration/java/com/liferay/commerce/inventory/engine/test/CommerceInventoryEngineTest.java

```diff
@@ -135,6 +303,213 @@ public void testConsumeQuantity() throws Exception {
 			commerceBookedQuantity.getCommerceInventoryBookedQuantityId());
 	}
 
+	@Test
+	public void testConsumeQuantityFromMultipleWarehouses() throws Exception {
+		frutillaRule.scenario(
+			"When the same warehouse item is added to 2 active warehouse the" +
+				"maximum consumable quantity is qual to the sum of the stock " +
+					"in both warehouses"
+		).given(
+			"2 active warehouse"
+		).and(
+			"A product sku in different quantities added to the warehouses"
+		).when(
+			"I retrieve the stock quantity after a consuming quantity is added"
+		).then(
+			"The stock quantity shall take into consideration only the " +
+				"remaining stock"
+		);
+
+		CommerceInventoryWarehouse commerceInventoryWarehouse1 =
+			CommerceInventoryTestUtil.addCommerceInventoryWarehouse(
+				_user.getGroupId());
+
+		CommerceInventoryWarehouse commerceInventoryWarehouse2 =
+			CommerceInventoryTestUtil.addCommerceInventoryWarehouse(
+				_user.getGroupId());
+
+		int warehouse1ItemQuantity = 5;
+		int warehouse2ItemQuantity = 5;
+
+		_commerceInventoryWarehouseItemLocalService.
+			addCommerceInventoryWarehouseItem(
+				_user.getUserId(),
+				commerceInventoryWarehouse1.getCommerceInventoryWarehouseId(),
+				_cpInstance1.getSku(), warehouse1ItemQuantity);
+
+		_commerceInventoryWarehouseItemLocalService.
+			addCommerceInventoryWarehouseItem(
+				_user.getUserId(),
+				commerceInventoryWarehouse2.getCommerceInventoryWarehouseId(),
+				_cpInstance1.getSku(), warehouse2ItemQuantity);
+
+		int companyStockQuantity = _commerceInventoryEngine.getStockQuantity(
+			_company.getCompanyId(), _cpInstance1.getSku());
+
+		Assert.assertEquals(
+			warehouse1ItemQuantity + warehouse2ItemQuantity,
+			companyStockQuantity);
+
+		int quantity = 7;
+
+		// consume quantity is called only after a validation done by the
+		// shipping process
+
+		_commerceInventoryEngine.consumeQuantity(
+			_user.getUserId(),
+			commerceInventoryWarehouse1.getCommerceInventoryWarehouseId(),
+			_cpInstance1.getSku(), quantity, 0, Collections.emptyMap());
+
+		int remainingCompanyStockQuantity =
+			_commerceInventoryEngine.getStockQuantity(
+				_company.getCompanyId(), _cpInstance1.getSku());
+
+		Assert.assertEquals(
+			warehouse1ItemQuantity + warehouse2ItemQuantity - quantity,
+			remainingCompanyStockQuantity);
+	}
+
+	@Test
+	public void testCreateMultipleWarehousesWithSameAttributes()
+		throws Exception {
+
+		frutillaRule.scenario(
+			"It should not be possible to create multiple warehouses with " +
+				"same attributes"
+		).given(
+			"One warehouse is created with external reference"
+		).when(
+			"Another warehouse with same attributes is create"
+		).then(
+			"An exception shall be raised"
```

Comment:
Is this actually what we're testing for here? I don't see an exception expected anywhere in this test.
EDIT: nvm, I see you fixed it in a later commit :)